game/src/ecs/components/HealthComponent.java

```diff
@@ -0,0 +1,178 @@
+package ecs.components;
+
+import ecs.damage.Damage;
+import ecs.damage.DamageType;
+import ecs.entities.Entity;
+import graphic.Animation;
+import java.util.ArrayList;
+import java.util.List;
+import semanticAnalysis.types.DSLContextMember;
+import semanticAnalysis.types.DSLType;
+import semanticAnalysis.types.DSLTypeMember;
+
+/** The HealthComponent makes an entity vulnerable and killable */
+@DSLType(name = "health_component")
+public class HealthComponent extends Component {
+    private static final List<String> missingTexture = List.of("animation/missingTexture.png");
+
+    private final List<Damage> damageToGet;
+    private @DSLTypeMember(name = "maximal_hit_points") int maximalHitPoints;
+    private int currentHitPoints;
+    private @DSLTypeMember(name = "on_death_function") IOnDeathFunction onDeath;
+    private @DSLTypeMember(name = "get_hit_animation") Animation getHitAnimation;
+    private @DSLTypeMember(name = "die_animation") Animation dieAnimation;
+
+    /**
+     * Creates a new HealthComponent
+     *
+     * @param entity associated entity
+     * @param maximalHitPoints maximum amount of hit-points, currentHitPoints cant be bigger than
+     *     that
+     * @param onDeath Function that gets called, when this entity dies
+     * @param getHitAnimation Animation to be played as the entity was hit
+     * @param dieAnimation Animation to be played as the entity dies
+     */
+    public HealthComponent(
+            Entity entity,
+            int maximalHitPoints,
+            IOnDeathFunction onDeath,
+            Animation getHitAnimation,
+            Animation dieAnimation) {
+        super(entity);
+        this.maximalHitPoints = maximalHitPoints;
+        this.currentHitPoints = maximalHitPoints;
+        this.onDeath = onDeath;
+        this.getHitAnimation = getHitAnimation;
+        this.dieAnimation = dieAnimation;
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Creates a HealthComponent with default values
+     *
+     * @param entity associated entity
+     */
+    public HealthComponent(@DSLContextMember(name = "entity") Entity entity) {
```

Comment:
Dieser Ctor sollte auf den "vollen" Ctor (Zeile 35ff) umleiten, d.h. nur einen Aufruf `this(entity, 1, 1, e -> {}, ...)` machen.